stdlib/Test/src/Test.jl

```diff
@@ -214,6 +227,25 @@ function Base.show(io::IO, t::Broken)
     end
 end
 
+# Types that appear in TestSetException.errors_and_fails we convert eagerly into strings
+# other types we convert lazily
+function Serialization.serialize(s::Serialization.AbstractSerializer, t::Pass)
```

Comment:
Is there any downside to stringifying everything eagerly so we don't need to add `serialize` methods?

Reply:
(I was sure I replied already—twice—but somehow my comments are gone?)

There's a possible performance concern (we almost never need these strings, but almost always need the other ones). More importantly though, there's a number of consumers that introspect the result of `Pass` to do further tests on the result.

Reply:
Hm, it's also possible that everything should be preserved as lazy until `serialize` (refs https://github.com/JuliaLang/julia/pull/24847)